rutil/ssl/SHA1Stream.cxx

```diff
@@ -18,27 +25,39 @@
 using namespace resip;
 
 SHA1Buffer::SHA1Buffer()
-        : mContext(new SHA_CTX()),
-          mBuf(SHA_DIGEST_LENGTH),
+        : mContext(NULL),
           mBlown(false)
 {
-   SHA1_Init(mContext.get());
-   setp(&mBuf[0], (&mBuf[mBuf.size()-1])+1);
+   std::memset(mBuf, 0, sizeof(mBuf));
+   setp(mBuf, mBuf + sizeof(mBuf));
+
+   mContext = EVP_MD_CTX_new();
+   if (!mContext)
+      throw std::bad_alloc();
+   int res = EVP_DigestInit_ex(mContext, EVP_sha1(), NULL);
+   if (!res)
+   {
+      EVP_MD_CTX_free(mContext);
+      mContext = NULL;
+      throw std::runtime_error("Failed to initialize SHA1 context");
+   }
 }
 
 SHA1Buffer::~SHA1Buffer()
 {
+   EVP_MD_CTX_free(mContext);
```

Comment:
For my code I created a deleter helper class:
```c++
struct OpenSslDeleter {
    void operator()(BIO* bio);
    void operator()(EVP_CIPHER_CTX* ctx);
    void operator()(EVP_ENCODE_CTX* ctx);
    void operator()(EVP_MD_CTX* ctx);
    void operator()(HMAC_CTX* ctx);
    void operator()(SSL* ssl);
    void operator()(SSL_CTX* ctx);
    void operator()(X509* x509);
};
```

With that I'm able to hold the OpenSSL contexts in RAII objects which enhances readability:
```c++
std::unique_ptr<EVP_CIPHER_CTX, OpenSslDeleter> m_ctx;
```

Reply:
Yes, but that requires C++11. Is libresiprocate requiring C++11?

Reply:
Yes, recent commits to the repo, now require C++ 11 for resip (ie: std::lock, std::mutex, etc).

Reply:
Ok, done.